src/ui/components/Search/Search.js

```diff
@@ -0,0 +1,168 @@
+ui.class.Search = class Search extends HTMLElement {
+	constructor(props) {
+		super();
+		const self = this;
+
+		// HTML tag variables
+		let m_parentTag;
+		let m_inputContainer;
+		let m_input;
+		let m_dropdownWrapperTag;
+		let m_dropdown;
+		let m_listbox;
+		let m_searchButton;
+
+		let m_props;
+		let m_value = "";
+
+		// Configure public methods
+		this.destroy = destroySearch;
+		this.setup = setup;
+		this.suggest = suggest;
+
+		if (props) {
+			setup(props);
+		}
+
+		async function setup(props) {
+			await configure(props);
+			await ui.utils.installUIComponent({ self, m_parentTag, m_props });
+			await setupDOM();
+			await setupInput();
+			await setupButton();
+			await setupDropdown();
+			setupComplete();
+		}
+
+		async function setupDOM() {
+			return new Promise((resolve) => {
+				m_inputContainer = ui.d.createTag({ ...m_props.tags.inputContainer, class: m_props.css.inputContainer });
+				self.classList.add(m_props.css.self);
+				self.appendChild(m_inputContainer);
+				resolve();
+			});
+		}
+
+		function setupInput() {
+			return new Promise((resolve) => {
+				let input = ui.utils.extend(true, {}, m_props.input);
+				input.css = ui.utils.extend(true, m_props.css.input, input.css);
+				input.parentTag = m_inputContainer;
+				input.fnClear = (context) => {
+					m_value = "";
+					if (m_dropdown) m_dropdown.close();
+				}
+				input.fnKeyup = (context) => {
+					if (m_props.input?.fnKeyup) {
+						m_value = context.value;
+
+						if (m_value.length >= m_props.firedIn) {
+							m_props.input.fnKeyup(m_value);
+						} else {
+							if (m_dropdown) m_dropdown.close();
+						}
+					}
+				}
+
+				m_input = ui.input(input);
+				resolve();
+			});
+		}
+
+		function setupButton() {
+			return new Promise((resolve) => {
+				let button = ui.utils.extend(true, {}, m_props.button);
+				button.css = ui.utils.extend(true, m_props.css.searchButton, button.css);
+				button.parentTag = self;
+				button.fnComplete = resolve;
+
+				button.fnClick = (context) => {
+					if (m_props.button?.fnClick && m_value?.length >= m_props.firedIn) {
+						m_props.button.fnClick(m_value);
+					}
+				}
+
+				m_searchButton = ui.button(button);
+			});
+		}
+
+		function setupDropdown() {
+			return new Promise((resolve) => {
+				if (m_props.suggest) {
+					m_dropdownWrapperTag = ui.d.createTag({ ...m_props.tags.wrapper, class: m_props.css.wrapper });
+					self.appendChild(m_dropdownWrapperTag);
+					let dropdown = ui.utils.extend(true, {}, m_props.dropdown);
+					dropdown.css = ui.utils.extend(true, m_props.css.dropdown, dropdown.css);
+
+					dropdown.fnComplete = (context) => {
+						installListbox(context.Dropdown);
+					}
+
+					dropdown.disableButton = true;
+					dropdown.positionTag = m_input;
+					dropdown.parentTag = m_dropdownWrapperTag;
+					m_dropdown = ui.dropdown(dropdown);
+				}
+				resolve();
```

Comment:
Move **resolve** function to prop **fnComplete** like it's done for the **Button** config